workflow/controller/operator.go

```diff
@@ -1731,7 +1731,9 @@ func (woc *wfOperationCtx) executeTemplate(ctx context.Context, nodeName string,
 			}
 		}
 
-		woc.updated = wfUpdated
+		if wfUpdated {
```

Comment:
The updated value is getting overwrite here.

Reply:
I assume you mean it is turned back into false? 

Minor suggestion:
```golang
woc.updated = woc.updated || wfUpdated
```